build(dependencies)📦: Add numpydoc to dependencies in pyproject.toml

ericmjl opened the pull request:
- Include numpydoc for documentation generation in utils.py.

Resolves #10735.

ericmjl committed:
build(dependencies)📦: Add numpydoc to dependencies in pyproject.toml

- Include numpydoc for documentation generation in utils.py.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10736) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10736) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10736) it.</sub>

ericmjl committed:
Update poetry lock file.

ericmjl commented:
@krrishdholakia @ishaan-jaff apologies for not tagging you guys earlier. Please let me know if there’s anything you believe I’ll need to do.

msabramo commented:
LGTM

krrishdholakia commented:
Hey @ericmjl since numpydoc is not required to make a .completion() request, I'm hesitant to add it as a package requirement. 

Is there a reason it needs to be a required dependency?

ericmjl commented:
Good point, @krrishdholakia! It becomes an issue when we use the function_to_dict function to convert a function into a tool schema. Is this planned to be supported long-term? If not, I’m happy to close the PR to save on deps and maintenance work.

krrishdholakia commented:
What would you want here @ericmjl 

This seems similar to how integrations might have their own dependencies but it's not part of the core requirements.

ericmjl commented:
I’m happy to follow your lead here. If I were to suggest, making it an optional dependency might be what I would go for.

krrishdholakia commented:
That sounds reasonable! Let's do that. 

Feel free to update your PR to reflect that. Happy to merge it in.

ericmjl committed:
chore(dependencies)🔧: Update optional dependencies and markers in poetry.lock and pyproject.toml

- Set several packages as optional in poetry.lock.
- Add 'extra == "utils"' marker to various packages in poetry.lock.
- Update numpydoc dependency to be optional in pyproject.toml.
- Add 'utils' extra section in pyproject.toml with numpydoc.

ericmjl commented:
@krrishdholakia can I check, are you ok with the current arrangement of deps, or would you prefer a different one?

ericmjl committed:
Merge branch 'main' into numpydoc-deps
